### PR TITLE
Create archive page asynchronously

### LINKED ISSUE #959: Speed up getting the number of files for a proposal
In order to get the number of files for a given proposal as listed on the instrument archive page, JWQL queries MAST for each program. These queries at the moment seem to be taking about 0.1 seconds per program. So as the number of programs goes up, so will the page loading time. 

But, eventually each program completes, at which point the number of files will no longer change. I wonder if there's a way to check the status of a program and once it is complete, we can store the number of files somewhere, so that we no longer need to query MAST for it?

### LINKED ISSUE #1157: Archive page for NIRCam is very slow to load
The NIRCam page is a lot slower to load than the pages for the other instruments, probably because there are more NIRCam proposals, and each proposal has more images, than those of the other instruments. It would be great if we could find a way to speed up the loading of this page.

One thought: Could this be related to the change we made a while back where we replaced a glob call over each proposal's list of preview images/thumbnails and replaced that with a text file that is read in? That sped things up at the time, but maybe now that text file is getting so big that reading it is taking a long time?

### PR DESCRIPTION
**Pull request by bhilbert4**
Rather than querying the database and constructing the context for an instrument's archive page at the time of the request, do that work asynchronously, and save the results in a json file. Then when there is a request for the page, read in the json file and create the page.

### CODE REVIEW

**Comment by bhilbert4** [comment]
Github won't let me link it, but this should also resolve #959

**Comment by bhilbert4** [comment]
For this to work on the servers, we need to create the os.path.join(get_config('outputs'), "archive_page")  directory.

**Comment by bhilbert4** [comment]
On the test server: 
Time needed to load the nircam archive page using the develop branch: 1 minute and 45 seconds
Time needed when using this branch: 13 seconds.
🥳

**Comment by bhilbert4** [comment]
@mfixstsci I think this is ready for review.

**Comment by bhilbert4** [comment]
With these changes, the json files containing the content of the archive pages are updated after each run of archive_database_update.py, which is done via cron job.

**Comment by bhilbert4** [comment]
In addition to creating the new archive_page subdirectory under the outputs directory, we'll also need to run the new script on all of the instruments when we merge this.

```
from datetime import datetime
from jwql.website.apps.jwql.data_containers import create_archived_proposals_context
insts = ['nircam', 'miri', 'nirspec', 'niriss', 'fgs']
for inst in insts:
   print(f'Working on {inst}, {datetime.now()}')
   create_archived_proposals_context(inst)
```

File: jwql/website/apps/jwql/views.py [hunk 1]
```diff
@@ -331,81 +332,13 @@ def archived_proposals_ajax(request, inst):
...
-    thumbnails_dict['proposals'] = proposal_nums
-    thumbnails_dict['thumbnail_paths'] = thumbnail_paths
-    thumbnails_dict['num_files'] = total_files
-    thumbnails_dict['viewed'] = proposal_viewed
-    thumbnails_dict['exp_types'] = thumb_exp_types
-    thumbnails_dict['obs_time'] = thumb_obs_time
+    # Read in the json file created by data_containers.create_archived_proposals_context
+    # and use as the context
+    output_dir = get_config()['outputs']
+    context_file = os.path.join(output_dir, 'archive_page', f'{inst}_archive_context.json')
```
Comment (mfixstsci) [inline jwql/website/apps/jwql/views.py:338]: This is slick 👍

**Review by mfixstsci — APPROVED** [review]
LGTM @bhilbert4 I am going to rebase the branch and then merge it once the CI is done.

**Pull request merged**